File: run/etc/templates/deviceServer/defaultCpp/src/__CLASS_NAME__.cc
/*
 * $Id$
 *
 * Author: <__EMAIL__>
 *
 * Created on __DATE__
 * 
 * Copyright (C) 2010-2012 European XFEL GmbH Hamburg. All rights reserved.
 */

#include <karabo/core/DeviceServer.hh>
#include <karabo/core/Runner.hh>

using namespace karabo::core;
using namespace karabo::util;
using namespace karabo::log;

int main(int argc, char** argv) {
  try {
      
      DeviceServer::Pointer deviceServer = Runner<DeviceServer>::instantiate(argc, argv);
      if (deviceServer) deviceServer->run();
      
  } catch (const karabo::util::TimeoutException& e) {
    std::cout << std::endl << "An error has occurred: Network response timed out." << std::endl;
    std::cout << "Make sure that a master-device-server is running under the configured broker/topic." << std::endl;
  } catch (const karabo::util::Exception& e) {
      std::cout << e;
  } catch (...) {
      RETHROW
  }
  return EXIT_SUCCESS;
}

File: run/etc/templates/pythonDevice/minimal/src/__PACKAGE_NAME__/server/__CLASS_NAME__.py
__author__ = "__EMAIL__"
__date__ = "__DATE__"
__copyright__ = """\
Copyright (c) 2010-2015 European XFEL GmbH Hamburg.
All rights reserved.
"""

from karabo.bound import runSingleDeviceServer


def main():
    # Note: One or more device class names can be passed in.
    runSingleDeviceServer('__CLASS_NAME__', serverId='__PACKAGE_NAME__-server')
